pkg/testingutil/devfile.go

```diff
@@ -108,7 +105,21 @@ func (d TestDevfileData) AddComponents(components []common.DevfileComponent) err
 
 func (d TestDevfileData) UpdateComponent(component common.DevfileComponent) {}
 
-func (d TestDevfileData) AddCommands(commands []common.DevfileCommand) error { return nil }
+func (d *TestDevfileData) AddCommands(commands ...common.DevfileCommand) error {
+	if d.Commands == nil {
+		d.Commands = make(map[string]common.DevfileCommand, 7)
+	}
+
+	for _, command := range commands {
+		id := command.GetID()
+		if _, ok := d.Commands[id]; !ok {
+			d.Commands[id] = command
+		} else {
+			panic("command with id '" + id + "' already exists in this TestDevfileData")
```

Comment:
Why panic here instead of returning a new error?

Reply:
Because it's not a recoverable error? Trying to add a command with an already existing id in the test data should fail. Is panicking causing undesired side-effects?

Reply:
I think it will cause the test suite to stop and the other tests won't run. Better to return and let the caller handle it.